pkg/controller/certificate-shim/sync_test.go

```diff
@@ -536,6 +536,105 @@ func TestSync(t *testing.T) {
 				},
 			},
 		},
+		{
+			Name:   "return a single HTTP01 Certificate for an ingress with a single valid TLS entry and valid secret template annotation",
+			Issuer: acmeClusterIssuer,
+			IngressLike: &networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-name",
+					Namespace: gen.DefaultTestNamespace,
+					Annotations: map[string]string{
+						cmapi.IngressClusterIssuerNameAnnotationKey: "issuer-name",
+						cmapi.IngressSecretTemplate:                 `{ "annotations": { "example-annotation" : "dummy-value" }, "labels": { "example-label" : "dummy-value" } }`,
```

Comment:
I think it will be easier to read and create these annotations if there was a separate annotation for SecretTemplateAnnotations and SecretTemplateLabels.